propolis/src/block/crucible.rs

```diff
@@ -35,21 +36,23 @@ impl CrucibleBackend {
         gen: u64,
         request: VolumeConstructionRequest,
         read_only: bool,
+        producer_registry: Arc<tokio::sync::Mutex<Option<ProducerRegistry>>>,
```

Comment:
Should be possible to just take `Option<ProducerRegistry>` here. It doesn't seem like we would want crucible to be modifying it behind the `Mutex`. See other comment re: Arc.

Reply:
Yep!  Thanks for your suggestions here.